main.py

```diff
@@ -109,54 +58,115 @@ def main(cfg: DictConfig):
+        metrics.run_llm_benchmark(base_model, logger=logger, tag="base_model")
+
+    base_fine_losses = train(
+        base_model,
         device,
         fine_train_loader,
         cfg.finetune.epochs,
         learning_rate=cfg.finetune.lr,
     )
-    _, model_fine_acc = test(fine_model, device, fine_test_loader)
+    wandb.log({"base_fine_losses": base_fine_losses})
 
-    for unadapt_config in cfg.unadapt:
-        ufm_pre_acc, ufm_fine_acc, ufm_fine_losses = calculate_unadaptability_metrics(
-            model,
-            unadapt_config,
+    if cfg.metric == "accuracy":
+        _, base_fine_acc = metrics.calculate_loss_and_accuracy(
+            base_model,
             device,
-            config,
-            pre_train_loader,
-            pre_test_loader,
-            fine_train_loader,
             fine_test_loader,
         )
-        pre_acc_ratio = ufm_pre_acc / pre_acc
-        fine_acc_ratio = ufm_fine_acc / model_fine_acc
-        loss_gap_ratio = calculate_loss_gap_ratio(ufm_fine_losses, model_fine_losses)
+    elif cfg.metric == "llm_tasks":
+        metrics.run_llm_benchmark(base_model, logger=logger, tag="base_model_finetuned")
+
+    # clean base model since we don't need it anymore
+    del base_model
+
+    logger.info("Running unadapt methods")
+    ufm_model = unadapt.apply_unadaptable_method(
+        model=ufm_model,
+        unadapt_method=cfg.unadapt.method,
+        config=OmegaConf.to_container(cfg.unadapt, resolve=True),
+        device=device,
+        train_loader=pre_train_loader,
+    )
+
+    logger.info("Benchmarking unadapted model for relative pre-training performance")
+    _, ufm_pre_acc = metrics.calculate_loss_and_accuracy(
+        ufm_model, device, pre_test_loader
+    )
+
+    if cfg.get("countermeasure", None) is not None:
+        logger.info("Running countermeasures")
+        ufm_model = countermeasures.run_countermeasures(
+            ufm_model, cfg.countermeasure, logger
+        )
+
+    # Finetune unadaptable model on finetuning dataset
+    ufm_fine_losses = train(
+        ufm_model,
+        device,
+        fine_train_loader,
+        cfg.finetune.epochs,
+        learning_rate=cfg.finetune.lr,
+    )
+    wandb.log({"ufm_fine_losses": ufm_fine_losses})
 
-        print(f"Unadaptable method: {unadapt_config}")
-        print(f"Model acc: {pre_acc}")
+    _, ufm_fine_acc = metrics.calculate_loss_and_accuracy(
+        ufm_model, device, fine_test_loader
+    )
+
+    if cfg.metric == "accuracy":
+        pre_acc_ratio = ufm_pre_acc / base_pre_acc
+        fine_acc_ratio = ufm_fine_acc / base_fine_acc
+        wandb.log(
+            {
+                "base_pre_acc": base_pre_acc,
+                "ufm_pre_acc": ufm_pre_acc,
+                "base_fine_acc": base_fine_acc,
+                "ufm_fine_acc": ufm_fine_acc,
+                "pretrain_acc_ratio": pre_acc_ratio,
+                "fine_acc_ratio": fine_acc_ratio,
+            }
+        )
```

Comment:
Same note about logging things in categories with slashes. That way, they show up in folders in wandb.